extensions/dim-rename/src/main/java/io/druid/query/extraction/ClusterDimRenameFn.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Function;
+import com.metamx.common.StringUtils;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ *
+ */
+@JsonTypeName("rename")
+public class ClusterDimRenameFn extends AbstractDimRenameFn implements DimExtractionFn
+{
+  private static final byte CACHE_KEY = 0x07;
+
+  private final String topic;
+
+  @JsonCreator
+  public ClusterDimRenameFn(
+      @JacksonInject final RenameManager renameManager,
```

Comment:
I would prefer if we did not add `@JacksonInject` annotations to objects that do the query serialization and de-serialization. This prevents someone from using the DimExtractionFn classes directly to construct Druid queries programmatically. Instead I think we should use an approach similar to what is done with DimFilters, where the DimFilter classes are just simple POJOs for serde purposes, and the actual implementation is implemented as a Filter classes.